Move UI plugin registration codes to vic-product

wjun opened the pull request:
Fixes [vmware/vic-product/#1432](https://github.com/vmware/vic-product/issues/1432)

zjs commented:
Does this PR take into consideration the notes in https://github.com/vmware/vic-product/issues/1432#issuecomment-412234184?

zjs commented:
> Fixes vmware/vic-product/#1432

I think this PR makes progress towards this issue, but does not fix it.

wjun commented:
@zjs Thanks for the reminder. I have changed the the feature branch and rebased &resolved the conflicts at https://github.com/vmware/vic/pull/8219 for review. I will abandon this pr.
